features/step_definitions/secrets.js

```diff
@@ -68,51 +41,38 @@ module.exports = function server() {
             Assert.equal(response.statusCode, 201);
 
             this.secretId = response.body.id;
-        })
-    );
+        });
+    });
 
-    this.When(/^the "main" job is started$/, { timeout: TIMEOUT }, () =>
-        request({
-            uri: `${this.instance}/${this.namespace}/pipelines/${this.pipelineId}/jobs`,
-            method: 'GET',
+    When(/^the "main" job is started$/, { timeout: TIMEOUT }, function step() {
+        return request({
+            uri: `${this.instance}/${this.namespace}/builds`,
+            method: 'POST',
+            body: {
+                jobId: this.jobId
+            },
+            auth: {
+                bearer: this.jwt
+            },
             json: true
-        }).then((response) => {
-            Assert.equal(response.statusCode, 200);
+        }).then((resp) => {
+            Assert.equal(resp.statusCode, 201);
 
-            this.jobId = response.body[0].id;
-            this.secondJobId = response.body[1].id;
-            this.thirdJobId = typeof response.body[2] === 'object' ? response.body[2].id : null;
-            this.lastJobId = response.body.reverse().find(b => typeof b === 'object').id || null;
-        })
-        .then(() =>
```

Comment:
Moved this bit to be part of the function that got moved to `world`, reason being that the `secondJobId` was set here, but used in a [step](https://github.com/screwdriver-cd/screwdriver/pull/617/files#diff-2b66eab6945aba4b87fba63045f30482R74) in a different scenario that ran afterwards.

In cucumber 2, the `world` object is rebuilt for every scenario, so it was undefined when it came time to actually use it.